### ISSUE TITLE
__dirname should not cause error when using node env

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/typescript-eslint-parser#46]
Hi, thanks for your amazing work. I have some feedback but I am very aware this project is still in the very early stages.

Versions:

```
$ npm ls eslint typescript-eslint-parser
/Users/OliverJAsh/Development/samefourchords.com
├── eslint@3.0.1
└── typescript-eslint-parser@0.1.0-alpha-1
```

ESLint config:

``` json
{
    "extends": "eslint:recommended",
    "parser": "typescript-eslint-parser",
    "env": {
        "node": true
    }
}
```

Input and output:

```
$ echo "console.log(__dirname)" > test.js
$ eslint test.js

/Users/OliverJAsh/Development/samefourchords.com/test.js
  1:1   error  Unexpected console statement  no-console
  1:13  error  '___dirname' is not defined   no-undef
```

Since I set the node env to true, I do not expect `__dirname` to cause an error.

**Comment by nzakas** [comment]
Sorry, same response as here: https://github.com/eslint/typescript-eslint-parser/issues/47#issuecomment-233098147

**Comment by jbcpollak** [comment]
Hi @nzakas - Its been quite a few months, has any progress been made about this? I read the comment you link to, but I don't understand quite how that issue relates to this one.

Is there a work around? Should we just do `// eslint-disable-line` for now?

**Comment by soda0289** [comment]
@jbcpollak The error with node globals such as __filename and __dirname has been fixed in master. Check the issue https://github.com/eslint/typescript-eslint-parser/issues/145 for more details.

When you enable a parser in eslint it will parse all files passed into it, this includes javascript and typescript files.

If you still have problems with the master branch you can open a new issue and we can look into.

**Comment by jbcpollak** [comment]
@soda0289 thanks - should I used a Git URL for the dependency or is it published to npm? Thanks!

**Comment by soda0289** [comment]
@jbcpollak We have one pull request, that fixes some JSX issues, that is waiting on a review. As soon as it is merged there will be a new release. Should not take too long but if you want to try out master today use the git url.
